seasmon_xr/ops/zonal.py

```diff
@@ -32,7 +32,7 @@ def do_mean(pixels, z_pixels, nodata, num_zones, dtype="float64"):
             for cl in range(nc):
                 pix = pixels[tix, rw, cl]
                 z_idx = z_pixels[rw, cl]
-                if pix != nodata and z_idx >= 0:
+                if (pix != nodata) and (z_idx != z_nodata):
```

Comment:
I guess this one assumes non-float data on input? 
Will need nan check for float32 inputs, but this will be probably be better off with a separate function for float data with appropriate routing in python

Reply:
Yes, it assumes zones as ints. Could we not add `!= np.nan` as an `or` condition in the second bracket, or are there unintended results if data comes in integer?

Reply:
I mean for `pix`. 

And no, you can't compare with `nan` directly, it's not equal to anything including itself. For `float32` data instead of `pix != nodata` one would have to have `not isnan(pix)`. But since `nodata` is a compulsory part of the interface the assumption must be that it's using int values.

Reply:
gotcha, yes that'll need to be handled better